src/components/Base/Input/input.jsx

```diff
@@ -12,31 +13,55 @@ export default class Input extends React.Component {
     disabled: PropTypes.bool,
     icon: PropTypes.string,
     iconSize: PropTypes.number,
-    iconType: PropTypes.string
+    iconType: PropTypes.string,
+    valueChange: PropTypes.func
   };
 
   static defaultProps = {
     className: '',
     icon: '',
     iconType: 'light',
     iconSize: 16,
+    valueChange: _.noop,
+    onChange: _.noop,
     disabled: false
   };
 
+  onChange = event => {
+    const { target } = event;
+    const value = target.type === 'checkbox' ? target.checked : target.value;
+    const name = target.name;
+
+    this.props.valueChange(name, value);
+    this.props.onChange(event);
```

Comment:
since you get target value, using `this.props.onChange(value)` is better

Reply:
```
  @action
  valueChange = (name, value) => {
    this.attribute[name] = value;
    if (this.attribute.name) {
      this.disableNextStep = false;
    } else {
      this.disableNextStep = true;
    }
    this.errorMessage = '';
  };
```

Reply:
valueChange pass the name param, I can use a generic function to change the store value